src/session/session_prefetch.c

```diff
@@ -10,54 +10,28 @@
 
 /*
  * __wt_session_prefetch_check --
- *     Check to see whether cursors owned by this session might benefit from doing pre-fetch
- *
- * FIXME-WT-11758 Change this function so that when pre-fetching is enabled on the session level,
- *     all cursors associated with that session with automatically perform pre-fetching. Use the
- *     session-level configuration once it is introduced.
+ *     Turn on prefetching for all cursors within a session if pre-fetching is configured on the
+ *     session.
  */
 bool
 __wt_session_prefetch_check(WT_SESSION_IMPL *session, WT_REF *ref)
 {
-    /* Internal threads should not be configured to do pre-fetching. */
-    if (!S2C(session)->prefetch_auto_on || F_ISSET(session, WT_SESSION_INTERNAL))
-        return (false);
-
     if (S2C(session)->prefetch_queue_count > WT_MAX_PREFETCH_QUEUE)
         return (false);
-
     /*
-     * Don't deal with internal pages at the moment - finding the right content to preload based on
-     * internal pages is hard.
+     * Check if pre-fetching is enabled on the session level. If not, don't perform pre-fetching on
+     * any of the associated cursors. We don't perform pre-fetching on internal threads or internal
+     * pages either (finding the right content to preload based on internal page is hard), so check
+     * for those here too.
```

Comment:
I'm not sure about the comment - it's not actually doing anything on cursors. Rather, it's expected that the cursor code will use this for prefetching. I don't have any great ideas for rewording it springing to mind - happy to chat about this offline.

Reply:
Yeah, I see your point about the comment - that's very fair. Will chat further offline.